hail/src/main/scala/is/hail/expr/ir/BlockMatrixWriter.scala

```diff
@@ -25,12 +25,12 @@ case class BlockMatrixNativeWriter(
   forceRowMajor: Boolean,
   stageLocally: Boolean) extends BlockMatrixWriter {
 
-  def apply(hc: HailContext, bm: BlockMatrix): Unit = bm.write(path, overwrite, forceRowMajor, stageLocally)
+  def apply(hc: HailContext, bm: BlockMatrix): Unit = bm.write(hc.sFS, path, overwrite, forceRowMajor, stageLocally)
```

Comment:
We need to start threading through FS to all function calls that require file system operations. This is a slightly dirty change, could be saved for a new PR if you'd like